### DISCUSSION TITLE
Signed in user cannot select tables when RLS is on

### DISCUSSION THREAD

**Discussion by peterscottsa** [supabase/supabase#9192] [Questions]
Hi there,

I am using the Supabase javascript client and Nextjs with magic links for auth. I followed the quick setup tutorial.

I have a basic RLS setup for a table called `teachers` that should allow only authenticated users (essentially the admins) to do any actions relating to the management of teachers. 

RLS: 
![Screenshot 2022-09-27 at 15 14 29](https://user-images.githubusercontent.com/1768648/192536285-d7e49c37-149b-44fd-86c2-afe78416c079.png)

This setup is very basic and there is no foreign key/ relations between these tables, its simply a way for me to put a password in front of the app.

![Screenshot 2022-09-27 at 15 16 31](https://user-images.githubusercontent.com/1768648/192536748-3bef9e7b-23d7-4c9d-b5b3-b067f73b7873.png)

The problem:
I can insert records, but I cannot select.

Here is an example of the select:
`export const getAllTeachers = async () =>
    await supabase.from('teachers').select()`
    
  If I disable RLS on the table this query works. My user is authenticated using the same method in the quick setup.

**Comment by silentworks** [comment]
Since you've enabled RLS on this table, by default until you add a policy it will deny access to any actions on this table. You will need to create a policy for **SELECT** action which has a value that will return true in order to select from the table.

**Reply by peterscottsa** [reply]
Thanks for the quick reply, as you can see in my question I have already specified a policy. I also tried a new policy that specifically only targets select actions, just to see if there is a difference:

<img width="1100" alt="Screenshot 2022-09-27 at 17 15 33" src="https://user-images.githubusercontent.com/1768648/192566162-1a183978-031c-4052-ba9a-7ba2e5baecaa.png">


Unfortunately this also doesn't seem to work.

**Reply by silentworks** [reply]
There might be an issue because you are doing an all then a specific on the same table. It's normally better to just do specific ones and test that out. So one for SELECT and one for INSERT.
